### PR TITLE
Add Libp2pChannel implementation of .try-receive-agreement method

### PR DESCRIPTION
**Pull request by AlexKnauth**

### CODE REVIEW

**Review by zenhack — COMMENTED** [review]
Some comments in-line. Also I am still uncomfortable with all of the copypasta in the integration tests; it's actually kindof hard to follow what the differences between all the tests are. I'm a bit torn  as to whether I think it's worth addressing now though, since I think we'll end up throwing a lot of this out anyway...

File: t/buy-sig-libp2p-both-generate-integrationtest.ss [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+       (with-io-port proc-seller
+         (lambda ()
+           (answer-questions
+             [["Choose your identity:"
+               (lambda (id) (string-prefix? "t/bob " id))]])))
+
+       (DBG "Filling up buyer prompt")
+
+       ;; Fill up buyer prompt
+       ;; TODO: negotiations on initial-block, check their within a range and then take the higher one
```
[the remark this answers is not in the document]
Comment (AlexKnauth) [reply t/buy-sig-libp2p-both-generate-integrationtest.ss:157]: It still applies, as when we do implement negotiations on initial-block, we will want to change this test to use that instead.

File: cli/interaction.ss [hunk 2]
```diff
@@ -442,8 +442,12 @@
...
         ;;  - yes: check match, FOR NOW: error on mismatch, TODO LATER: possible negotiations on failure
         (other-agreement 
-         (unless (equal? agreement other-agreement)
-           (error "start-interaction/generate-agreement: generated agreement doesn't match received agreement"))
+         (unless (equal? (json<- InteractionAgreement agreement)
+                         (json<- InteractionAgreement other-agreement))
+           (DDT "agreements-mismatch: generated agreement doesn't match received agreement"
+             InteractionAgreement agreement
+             InteractionAgreement other-agreement)
+           (error "start-interaction/generate-agreement: generated agreement doesn't match received agreement" agreement other-agreement))
```
[the remark this answers is not in the document]
Comment (AlexKnauth) [reply cli/interaction.ss:450]: The `equal?` check was not working before.

File: t/buy-sig-libp2p-without-wait-for-agreement-integrationtest.ss [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+      (def boot-addr (peer-info->string (libp2p-identify boot-c))) ;;get the addr of boot-c
+
+      ;; FIXME: Test for polling
+      ;; If we try to poll when running the buy_sig interaction via cli,
+      ;; it works just fine.
+      ;; In integration tests however, if we start the buyer's interaction first,
+      ;; and poll until seller comes online, we get the following error message
+      ;; when sending the agreement over the channel:
+      ;;
+      ;;   ERROR IN vyzo/libp2p/client#libp2p-stream__% -- libp2p-request: [libp2p-error] protocol not supported
```
[the remark this answers is not in the document]
Comment (AlexKnauth) [reply t/buy-sig-libp2p-without-wait-for-agreement-integrationtest.ss:87]: Oh, that comment may be outdated. I'll look into it, and either delete it or update it.

[the remark this answers is not in the document]
Comment (AlexKnauth) [reply t/buy-sig-libp2p-without-wait-for-agreement-integrationtest.ss]: I have removed the comment in both places, and added a new test case to check that either order of starting buyer/seller works.

**Review by zenhack — APPROVED** [review]
